library/ziti.c

```diff
@@ -439,3 +724,160 @@ static void version_cb(ctrl_version *v, ziti_error *err, void *ctx) {
         free_ctrl_version(v);
     }
 }
+
+#define OID_PKCS7 MBEDTLS_OID_PKCS "\x07"
+#define OID_PKCS7_DATA OID_PKCS7 "\x02"
+#define OID_PKCS7_SIGNED_DATA OID_PKCS7 "\x01"
+
+static void well_known_certs_cb(char *base64_encoded_pkcs7, ziti_error *err, void *req) {
+    struct nf_enroll_req *enroll_req = req;
+    size_t der_len;
+    unsigned char *p;
+    unsigned char *end;
+    mbedtls_x509_crt *cp;
+    mbedtls_x509_crt certs={0};
+    unsigned char *cert_buf;
+
+    int rc = mbedtls_base64_decode( NULL, 0, &der_len, base64_encoded_pkcs7, strlen(base64_encoded_pkcs7)); // determine necessary buffer size
+    char *base64_decoded_pkcs7 = calloc(1, der_len + 1);
+    rc = mbedtls_base64_decode( base64_decoded_pkcs7, der_len, &der_len, base64_encoded_pkcs7, strlen(base64_encoded_pkcs7));
+
+    unsigned char *der = (unsigned char*)base64_decoded_pkcs7;
+
+    p = der;
+    end = der + der_len;
+    size_t len;
+    mbedtls_asn1_get_tag(&p, end, &len, MBEDTLS_ASN1_CONSTRUCTED | MBEDTLS_ASN1_SEQUENCE);
```

Comment:
we should check the return status of asn1 parsing functions

Reply:
done